controller/fsm.py

```diff
@@ -0,0 +1,109 @@
+"""Module that implements `drunc` finite state machine."""
+
+from statemachine import State, StateMachine
+
+from . import controller_interface as ci
+
+
+class DruncFSM(StateMachine):
```

Comment:
I'm still unsure what we'll need this class for once we have the controller backend. It's a cute way to define things but my impression is we only need to store a mapping of state names to transitions and transitions to targets.